File: backend/application/comic_service.py
from core.ports import ComicRepository
from core.entities import Comic

class ComicService:
    # Inyectamos la dependencia a través del constructor
    def __init__(self, repository: ComicRepository):
        self.repository = repository

    def obtener_catalogo(self):
        return self.repository.obtener_todos()

    def obtener_detalle(self, comic_id: int):
        comic = self.repository.obtener_por_id(comic_id)
        if not comic:
            # Aquí está nuestra regla de negocio: si no existe, lanzamos error
            raise Exception("Cómic no encontrado") 
        return comic

    def buscar_comics(self, titulo: str):
        return self.repository.buscar_por_titulo(titulo)

File: frontend/js/application/comicService.js
export class ComicService {
    // Inyectamos el adaptador de la API
    constructor(apiAdapter) {
        this.api = apiAdapter;
    }

    async obtenerCatalogo() {
        return await this.api.get('/comics');
    }

    async buscar(termino) {
        return await this.api.get(`/comics/buscar/?q=${termino}`);
    }

    async obtenerDetalle(id) {
        return await this.api.get(`/comics/${id}`);
    }
}